packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,710 @@
+      Properties: cdk.objectToCloudFormation(property.properties),
+      Configurations: cdk.listMapper(this.ApplicationConfigPropertyToJson)(property.nestedConfig),
+    };
+  }
+
+  private renderTags(tags: { [key: string]: any } | undefined): { [key: string]: any } {
+    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
+  }
+
+  /**
+   * Creates a Cloudwatch log group
+   *
+   * @returns name if user enables props.monitoring.logging to be true
+   **/
+  private renderLogGroupName(): string {
+    return new logs.LogGroup(this, 'Emr Containers Default Cloudwatch Log Group').logGroupName;
+  }
+
+  /**
+   * Creates a S3 bucket
+   *
+   * @returns an automatically generated bucket name
+   **/
+  private renderS3BucketName(): string {
+    return new s3.Bucket(this, 'Emr Containers Default S3 Bucket').bucketName;
+  }
+
+  // https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/creating-job-execution-role.html
+  private createJobExecutionRole(): iam.Role {
+    let jobExecutionRole: iam.Role = new iam.Role(this, 'Job-Execution-Role', {
+      assumedBy: new iam.ServicePrincipal('emr-containers.amazonaws.com'),
+    });
+
+    // allow job-execution-role to be used by Step Functions
+    jobExecutionRole.assumeRolePolicy?.addStatements(
+      new iam.PolicyStatement({
+        actions: ['sts:AssumeRole'],
+        principals: [new iam.ServicePrincipal('states.amazonaws.com')],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Managing Virtual Clusters Policies', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:CreateVirtualCluster',
+              'emr-containers:DeleteVirtualCluster',
+              'emr-containers:DescribeVirtualCluster',
+              'emr-containers:ListVirtualClusters',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers S3 and Log Policies', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              's3:PutObject',
+              's3:GetObject',
+              's3:ListBucket',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['arn:aws:logs:*:*:*'],
+            actions: [
+              'logs:PutLogEvents',
+              'logs:CreateLogStream',
+              'logs:DescribeLogGroups',
+              'logs:DescribeLogStreams',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Job Run Policy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
```

Comment:
add the virtual cluster ID if it's not a JSON path

Reply:
Removed because redundant and out of scope